### PR TITLE
feat: Support Delete Message API

### PR DESCRIPTION
**Pull request by kappa-lab**
# Describe the change
- support Delete Message API
- No breaking changes

# OpenAI documentation 
relevant API doc from https://platform.openai.com/docs/api-reference/messages/deleteMessage

# Additional context
- fix lint

### CODE REVIEW

File: messages_test.go [hunk 1]
```diff
@@ -8,20 +8,17 @@ import (
...
 	"github.com/sashabaranov/go-openai"
+	"github.com/sashabaranov/go-openai/internal/test"
 	"github.com/sashabaranov/go-openai/internal/test/checks"
 )
 
 var emptyStr = ""
 
-// TestMessages Tests the messages endpoint of the API using the mocked server.
-func TestMessages(t *testing.T) {
+func registerServer(t *testing.T, server *test.ServerTest) {
```
Comment (kappa-lab) [inline messages_test.go:17]: split func for linter

Comment (sashabaranov) [reply messages_test.go:17]: The `registerServer` name is very generic, let's name it with something more concrete for this test

**Comment by sashabaranov** [comment]
Thank you for the PR! Added a minor comment

File: messages_test.go (hunk 1 above)
Comment (kappa-lab) [reply messages_test.go]: rename to `setupServerForTestMessage` 
[8facd6e](https://github.com/sashabaranov/go-openai/pull/799/commits/8facd6e65b6b08056aa32b20db0b5b7e38fbd932)

**Review by sashabaranov — APPROVED** [review]

**Pull request merged**